components/camel-resteasy/src/main/java/org/apache/camel/component/resteasy/servlet/ResteasyServletInputStreamCopier.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.resteasy.servlet;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ReadListener;
+import javax.servlet.ServletInputStream;
+
+/**
+ * Class for copying input stream from HttpRequest
+ */
+public class ResteasyServletInputStreamCopier extends ServletInputStream {
+    
+    private InputStream input;
+    private ByteArrayOutputStream copy;
+
+    public ResteasyServletInputStreamCopier(InputStream inputStream) {
+        /* create a new input stream from the cached request body */
+        this.input = inputStream;
+        this.copy = new ByteArrayOutputStream();
```

Comment:
This will read the entire stream into memory.

Camel has a stream caching feature. Maybe we should revisit this later.

Reply:
https://issues.apache.org/jira/browse/CAMEL-15042